### ISSUE TITLE
paper_drawer_panel not working on firefox, drawer always closed

### ISSUE THREAD

**Issue by alextekartik** [dart-lang/polymer_elements#93]
In my app, the drawer is always closed. Trying to reproduce it in a simple example below (based on the javascript documentation), the drawer was shown below the main panel in firefox

The same example works fine on Chrome. There was a similar issue in version 1.0.3 of the js version due to a vulcanize issue: https://github.com/PolymerElements/paper-drawer-panel/issues/78. However the javascript version works fine (in the demo page)

``` html
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>paper_drawer_panel_example</title>
    <script src="packages/web_components/webcomponents-lite.min.js"></script>
    <link rel="import" href="packages/polymer_elements/paper_drawer_panel.html">
    <link rel="import" href="packages/polymer_elements/paper_header_panel.html">
    <link rel="import" href="packages/polymer_elements/paper_toolbar.html">
    <link rel="import" href="packages/polymer_elements/iron_icons.html">
    <link rel="import" href="packages/polymer_elements/paper_icon_button.html">
</head>
<body unresolved>
<paper-drawer-panel>
    <paper-header-panel drawer>
        <paper-toolbar>
            <div>Application</div>
        </paper-toolbar>
        <div> Drawer content...</div>
    </paper-header-panel>
    <paper-header-panel main>
        <paper-toolbar>
            <paper-icon-button icon="menu" paper-drawer-toggle></paper-icon-button>
            <div>Title</div>
        </paper-toolbar>
        <div> Main content...</div>
    </paper-header-panel>
</paper-drawer-panel>
<script type="application/dart" src="paper_drawer_panel_example.dart"></script>
<script src="packages/browser/dart.js"></script>
</body>
</html>
```

**Comment by dam0vm3nt** [comment]
With latest polymer on firefox and basically the same example I get a completely screwed up layout :

![schermata 2015-10-15 alle 14 16 35](https://cloud.githubusercontent.com/assets/2409466/10513361/68747e26-7347-11e5-804e-6858006f3f2a.png)

Examining with the inspector and comparing with chrome I can see there are missing styles.

**Comment by jakemac53** [comment]
Seems like maybe its related to the `vulcanize` issue that was linked there. I don't think that we are inlining external stylesheets.

**Comment by dam0vm3nt** [comment]
same problem with safari. Confirmed with `polymer_interop` rc.2.
Don't know if it is related but on firefox web console also there are a lot of warning about "unreachable code" .

**Comment by zoechi** [comment]
I encountered the unreachable code message in Firefox in another context. https://developer.mozilla.org/de/docs/Web/JavaScript/Reference/Statements/return, https://bugzilla.mozilla.org/show_bug.cgi?id=1151931
Seems to be caused by dart2js not adding `;` in the generated JS to reduce code size.

**Comment by dam0vm3nt** [comment]
confirmed with polymer-dart rc.5, polymer_interop rc.4, polymer_element rc.3

**Comment by dam0vm3nt** [comment]
_WORKAROUND FOUND_ : adding this line to your `index.html` will make things work better even though it is not perfect.

``` html
<link rel="stylesheet" href="packages/polymer_elements/src/paper-drawer-panel/paper-drawer-panel.css">

```

**Comment by zoechi** [comment]
The problem is then probably that `paper-drawer-panel.css` should be made a style-module and referenced here https://github.com/dart-lang/polymer_elements/blob/80101edba099447584412cf8d58ba4c8ebb716e9/lib/src/paper-drawer-panel/paper-drawer-panel.html#L110 like 

``` html
<dom-module id="paper-drawer-panel">
  <template>
    <style include="paper-drawer-panel-styles">
```

**Comment by dam0vm3nt** [comment]
I've noted that in the latest `polymer-js` sources there is no more such a `.css` and style is now inlined for `paper-drawer-panel`.

So I've cloned `polymer_elements`, updated version of `paper_elements` and run again apigen tool : with this version `paper-drawer-panel` seams to work ok in firefox too. So probably this was a `polymer-js` problem after all (although the polymer JS demo works well ...), or the version of elements was misaligned with the `polymer.js` in `polymer-dart` package (this could explain also why the js demo works) .

I took a rapid look and everything seams to work fine even though there are still some problems with `paper-dialog` not rendering correctly in firefox.

You can find my "updated" version here : 

https://github.com/dam0vm3nt/polymer_elements/tree/update_paper_eleme

I'm sending a PR too.

**Comment by dam0vm3nt** [comment]
I found that other components suffers from the same problem : 
- `paper-header-panel`
- `paper-dialog`

The common pattern is that they are using styles imported from external `.css` files. Inlining styles in components fix the problem but in some cases I had to modify the original polymer-JS sources (see [this](https://github.com/dam0vm3nt/polymer_elements/commit/4e69c963df912b2bb2062bac52d5b312805b5a41) and [this](https://github.com/dam0vm3nt/polymer_elements/commit/276cb0447b650eb00cbe24aa75f8e60be47694fe) for example).

Most probably the real problem is that polymer-dart doesn't handles correctly external (deprecated) stylesheet imports.
